### PR TITLE
skip "event" dims in log likelihood group whenever necessary

### PR DESCRIPTION
**Pull request by OriolAbril**
## Description
ArviZ schema recommends to use the same variable names in log likelihood group as the one in observed_data and posterior_predictive groups. This helps identifying to what terms does each pointwise log lik value correspond, and eases calculations like LOO-PIT. 

However, log likelihood and posterior predictive don't always have the same dimensions and shape. In multidimensional observations the _event_ dimensions (shape of a single _observation_) are reduced. The shape of posterior predictive is `(chain, draw, *batch_shape, *event_shape)` whereas for log likelihood it is `(chain, draw, *batch_shape)`, in scalar observations they are both the same, but for multidimensional observations such as multivariate normals or multinomials the two shapes are different and the converters error out.

This will automatically handle this event_shape difference _provided that they follow this convention_, that is, event dims are the last ones.

cc @AlexAndorra 

## Checklist

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#pull-request-checklist) PR format
- [x] Includes new or updated tests to cover the new feature
- [x] Code style  correct (follows pylint and black guidelines)
- [x] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

File: arviz/data/base.py [hunk 1]
```diff
@@ -189,6 +208,10 @@ def dict_to_dataset(data, *, attrs=None, library=None, coords=None, dims=None):
     dims : dict[str] -> list[str]
         Dimensions of each variable. The keys are variable names, values are lists of
         coordinates.
+    skip_event_dims : bool
+        If True, cut extra dims whenever present to match the shape of the data.
+        Necessary for PPLs who have the same name in both observed data and log
+        likelihood groups share variable names to account for their different shapes.
```
Comment (AlexAndorra) [inline arviz/data/base.py:214]:
```suggestion
        Necessary for PPLs which have the same name in both observed data and log
        likelihood groups, to account for their different shapes when observations are 
        multivariate.
```

**Review by AlexAndorra — CHANGES_REQUESTED** [review]
Thanks @OriolAbril, LGTM! I just spotted a doctring that I think could be more explicit, and still have to check that my PyMCon model works with this change now -- will keep you posted

**Comment by OriolAbril** [comment]
This should be ready to merge

**Review by AlexAndorra — APPROVED** [review]
This looks good to me, thanks @OriolAbril ! 
I haven't had time to test it on my PyMCon model yet (as I'm still using a fork of the repo, checking out PRs is always a bit of a pain; I should just clone the actual repo now FWIW) but as you did it on the Labs code, I suggest we merge and then I'll test when it's on the master branch?
Also, why is Azure not showing up here?

**Comment by OriolAbril** [comment]
> I haven't had time to test it on my PyMCon model yet (as I'm still using a fork of the repo, checking out PRs is always a bit of a pain; I should just clone the actual repo now FWIW) but as you did it on the Labs code, I suggest we merge and then I'll test when it's on the master branch?

I added a test on a pymc3 multinomial to be extra sure, the rest of the tests are backend agnostic at a lower level, so don't worry about it. For future reference, to quickly try the code of a PR (provided it's a PR created from a branch of the main repo) you can do `pip install git+https://github.com/arviz-devs/arviz@<branch-name>`, you won't be able to edit the source code and push back, but for quick testing is quite convenient.

> Also, why is Azure not showing up here?

No idea, we had some issues with Azure last week, hopefully they will have solved themselves by now (we'll need a new PR to see)

**Comment by AlexAndorra** [comment]
Sounds good! I'll go ahead and merge then

**Pull request merged**